base/Makefile

```diff
@@ -71,6 +71,7 @@ endif
 	@printf "%s\n" "const PRIVATE_LIBDIR = "$(call shell_escape,$(call julia_escape,$(call normalize_path,$(private_libdir_rel)))) >> $@
 	@printf "%s\n" "const PRIVATE_LIBEXECDIR = "$(call shell_escape,$(call julia_escape,$(call normalize_path,$(private_libexecdir_rel)))) >> $@
 	@printf "%s\n" "const INCLUDEDIR = "$(call shell_escape,$(call julia_escape,$(call normalize_path,$(includedir_rel)))) >> $@
+	@printf "%s\n" "const SOURCEDIR = "$(call shell_escape,$(call julia_escape,$(call normalize_path,$(JULIAHOME)))) >> $@
```

Comment:
AFAICT this new constant is needed as paths to source files are those at build time. `BUILD_ROOT_PATH` which was defined below was incorrect in claiming that it pointed to the build-time source path as it was based on `BINDIR`, yet there's no guarantee that the julia executable during build time is next to the source dir.